### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by mwellick**

### CODE REVIEW

File: app/cinema/hall.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+class CinemaHall:
+
+    def __init__(self, number: int) -> None:
+        self.number = number
+
+    def movie_session(self,
+                      movie_name: str,
+                      customers: list["Customer"],
+                      cleaning_staff: "Cleaner") -> None:
+        print(f'\"{movie_name}\" started in hall number {self.number}.')
```
Comment (EdAlekseiev) [inline app/cinema/hall.py:14]:
```suggestion
        print(f'"{movie_name}" started in hall number {self.number}.')

```

File: app/cinema/hall.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+        for customer in customers:
+            customer.watch_movie(movie_name)
+
+        print(f'\"{movie_name}\" ended.')
```
Comment (EdAlekseiev) [inline app/cinema/hall.py:19]:
```suggestion
        print(f'"{movie_name}" ended.')

```

File: app/cinema/hall.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+
+class CinemaHall:
+
+    def __init__(self, number: int) -> None:
+        self.number = number
+
+    def movie_session(self,
+                      movie_name: str,
+                      customers: list["Customer"],
```
Comment (EdAlekseiev) [inline app/cinema/hall.py:12]:
```suggestion
                      customers: list[Customer],

```

File: app/cinema/hall.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+                      cleaning_staff: "Cleaner") -> None:
```
Comment (EdAlekseiev) [inline app/cinema/hall.py:13]:
```suggestion
                      cleaning_staff: Cleaner) -> None:

```

File: app/cinema/bar.py [hunk 5]
```diff
@@ -0,0 +1,7 @@
+from app.people.customer import Customer
+
+
+class CinemaBar:
+    @staticmethod
+    def sell_product(product: str, customer: "Customer") -> None:
```
Comment (EdAlekseiev) [inline app/cinema/bar.py:6]:
```suggestion
    def sell_product(product: str, customer: Customer) -> None:

```

**Review by EdAlekseiev — CHANGES_REQUESTED** [review]

File: app/people/customer.py [hunk 6]
```diff
@@ -0,0 +1,8 @@
+class Customer:
+
+    def __init__(self, name: str, food: str) -> None:
+        self.name = name
+        self.food = food
+
+    def watch_movie(self, movie: str) -> None:
+        print(f'{self.name} is watching \"{movie}\".')
```
Comment (YukihiroSM) [inline app/people/customer.py:8]:
```suggestion
        print(f'{self.name} is watching "{movie}".')

```

File: app/main.py [hunk 7]
```diff
@@ -1,6 +1,40 @@
...
+from app.people.customer import Customer
+from app.people.cinema_staff import Cleaner
+from app.cinema.hall import CinemaHall
+from app.cinema.bar import CinemaBar
 
 
-def cinema_visit(customers: list, hall_number: int, cleaner: str, movie: str):
-    # write you code here
-    pass
+def cinema_visit(customers: list,
```
Comment (YukihiroSM) [inline app/main.py:8]: Add more definitive typehint, please. Like list[int] or something else in your case.

**Review by YukihiroSM — CHANGES_REQUESTED** [review]
Overall the code is ok, but fix some comments, please.

File: app/main.py [hunk 8]
```diff
@@ -1,6 +1,40 @@
...
+    {"name": "Bob", "food": "Coca-cola"},
+    {"name": "Alex", "food": "popcorn"}
+]
+hall_number = 5
+cleaner_name = "Anna"
+movie = "Madagascar"
+cinema_visit(customers=customers,
+             hall_number=5,
+             cleaner="Anna",
+             movie_name="Madagascar")
```
Comment (Bohdan-Salamakha) [inline app/main.py:40]: remove this test block of code

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]

**Review by Bohdan-Salamakha — APPROVED** [review]

**Pull request closed without merging**